ctfo_server.h

```diff
@@ -1116,27 +1140,88 @@ class CTFOServer final {
   }
 
  private:
+  struct PushNotificationsSender {
+    Storage& storage;
+    const std::chrono::microseconds starting_from;
+    current::integrations::onesignal::IOSPushNotificationsSender transport;
+
+    using transaction_t = typename Storage::transaction_t;
+
+    PushNotificationsSender(Storage& storage,
+                            std::chrono::microseconds last_pushed_notification_timestamp,
+                            const std::string& onesignal_app_id,
+                            uint16_t onesignal_local_port)
+        : storage(storage),
+          starting_from(last_pushed_notification_timestamp + std::chrono::microseconds(1)),
+          transport(onesignal_app_id, onesignal_local_port) {
+      std::cerr << "Starting sending push notifications from " << starting_from.count() << " epoch us.\n";
+    }
+
+    current::ss::EntryResponse operator()(const transaction_t& entry, idxts_t current, idxts_t) const {
+      if (current.us >= starting_from) {
+        for (const auto& mutation : entry.mutations) {
+          if (Exists<Persisted_NotificationUpdated>(mutation)) {
+            const UID uid = Value<Persisted_NotificationUpdated>(mutation).data.uid;
+            const std::string player_id =
+                Value(storage.ReadOnlyTransaction([uid](ImmutableFields<Storage> fields) -> std::string {
+                  const auto rhs = fields.uid_player_id[uid];
+                  if (Exists(rhs)) {
+                    return Value(rhs).player_id;
+                  } else {
+                    return "";
+                  }
+                }).Go());
+            if (!player_id.empty()) {
+              // Hack: Update `push_starting_from` on success of one push, leave intact on failure.
+              if (transport.Push(player_id, 1)) {
```

Comment:
`Push` can throw, but only if things go really bad, in which case I'd like the binary to crash.